lib/repositories/message.js

```diff
@@ -394,20 +412,89 @@ function createApi(table, provider, sqlGenerator, dependencies) {
                 });
             });
         });
+    },
+
+    /**
+     * Deletes all messages belonging to the specific mailbox
+     */
+    removeByMailbox: function remove(mailbox) {
+      dependencies.logger.trace('message.removeByMailbox called');
+
+      var query = table
+        .select(table.star())
+        .from(table)
+        .where(table['mailbox_id'].equals(mailbox.getId()))
+        .toQuery();
+      query = provider.forUpdate(query);
+
+      return provider.beginTransaction(true)
+        .then(function(transaction) {
+
+          return transaction.runQuery(query)
+            .then(function(result) {
+              var query = table
+                .delete()
+                .where(table['mailbox_id'].equals(mailbox.getId()))
+                .toQuery();
+
+              var messages = result.rows;
+
+              var x;
+              for (x in messages) {
+                messages[x] = convertFromStorage(messages[x]);
+              }
+
+              return transaction.runQuery(query)
+                .then(function() {
+                  return transaction.commit();
+                })
+                .then(function() {
+                  dependencies.logger.debug({messages: messages},
+                                            'Messages removed');
+                  return messages;
```

Comment:
Just remembered I suggested this. I think you mentioned that since the mailbox is being deleted, the MWI mailbox can just be delete via ARI by the caller. I think that would be best.